migration/20170626-1-move-third-party-config-to-external-integrations.py

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env python
+"""Move integration details from the Configuration file into the
+database as ExternalIntegrations
+"""
+import os
+import sys
+import json
+import logging
+from nose.tools import set_trace
+
+bin_dir = os.path.split(__file__)[0]
+package_dir = os.path.join(bin_dir, "..")
+sys.path.append(os.path.abspath(package_dir))
+
+from config import Configuration
+from core.model import (
+    ConfigurationSetting,
+    ExternalIntegration as EI,
+    Library,
+    get_one_or_create,
+    production_session,
+)
+
+log = logging.getLogger(name="Metadata Wrangler configuration import")
+
+def log_import(integration_or_setting):
+    log.info("CREATED: %r" % integration_or_setting)
+
+
+_db = production_session()
+try:
+    Configuration.load()
+
+    # Create the Bibblio integration.
+    bibblio_conf = Configuration.integration('Bibblio')
+    if bibblio_conf:
+        bibblio = EI(
+            name=EI.BIBBLIO,
+            protocol=EI.BIBBLIO,
+            goal=EI.METADATA_GOAL
+        )
+        _db.add(bibblio)
+        bibblio.username = bibblio_conf.get('client_id')
+        bibblio.password = bibblio_conf.get('client_secret')
+        log_import(bibblio)
+
+    # Create the Metadata Wrangler configuration.
+    metadata_wrangler_conf = Configuration.integration('Metadata Wrangler')
+    if metadata_wrangler_conf:
+        wrangler = EI(
+            name=EI.METADATA_WRANGLER,
+            protocol=EI.METADATA_WRANGLER,
+            goal=EI.METADATA_GOAL
+        )
+        _db.add(wrangler)
+        wrangler.url = metadata_wrangler_conf.get('url')
+        wrangler.username = metadata_wrangler_conf.get('client_id')
+        wrangler.password = metadata_wrangler_conf.get('client_secret')
+        log_import(wrangler)
+
+    # Get the base url.
+    content_server_conf = Configuration.integration('Content Server')
+    if content_server_conf:
+        url = content_server_conf.get('url')
+        setting = ConfigurationSetting.sitewide(_db, Configuration.BASE_URL_KEY)
+        setting.value = url
+        log_import(setting)
+
+    # Create a default library.
+    library = Library.instance(_db)
```

Comment:
This needs to change since `Library.instance` was removed today. Take a look at circulation, 20170224-copy-collection-configuration-into-database.py for some code to copy.